### PR TITLE
nvdimm_redis: Add new case to run redis test

### PR DESCRIPTION
**Pull request by huangyum**
Signed-off-by: Yumei Huang <yuhuang@redhat.com>
ID: 1906666

### CODE REVIEW

**Comment by huangyum** [comment]
(1/1) type_specific.io-github-autotest-qemu.nvdimm.nvdimm_redis: PASS (695.75 s)

@yanan-fu @maxujun Would you pls help review, thanks.

File: qemu/tests/nvdimm_redis.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    2) Run redis test inside guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    error_context.context("Install dependency packages in guest", logging.info)
```
Comment (yanan-fu) [inline qemu/tests/nvdimm_redis.py:19]: Miss logging module.

File: qemu/tests/nvdimm_redis.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        error_context.context("Get redis in guest", logging.info)
+        cmds = []
+        cmds.append(params["get_redis"])
+        cmds.append(params["get_nvml"])
+        cmds.append(params["compile_nvml"])
+        cmds.append(params["compile_redis"])
+        for cmd in cmds:
+            s, o = session.cmd_status_output(cmd, timeout=300)
+            if s:
+                test.error("Failed to run cmd '%s', status: %s, output: %s" %
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/nvdimm_redis.py:33]: Thanks, updated.

File: qemu/tests/nvdimm_redis.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/nvdimm_redis.py:19]: It's imported at L1.

File: qemu/tests/nvdimm_redis.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (yanan-fu) [inline qemu/tests/nvdimm_redis.py:34]: I am thinking if it is needed to show the error code `s`, not very necessary, how do you think about remove it ?

Comment (huangyum) [reply qemu/tests/nvdimm_redis.py:34]: No big deal to me. Any other comments? I will update together.

**Comment by huangyum** [comment]
@maxujun @MiriamDeng Could any one help review and provide test result on ppc host? Thanks.

**Comment by MiriamDeng** [comment]
OK, I will take a look on it, thanks.

File: qemu/tests/nvdimm_redis.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        test.cancel("Install dependency packages failed")
+    try:
+        error_context.context("Get redis in guest", logging.info)
+        cmds = []
+        cmds.append(params["get_redis"])
+        cmds.append(params["get_nvml"])
+        cmds.append(params["compile_nvml"])
+        cmds.append(params["compile_redis"])
+        for cmd in cmds:
+            s, o = session.cmd_status_output(cmd, timeout=300)
```
Comment (MiriamDeng) [inline qemu/tests/nvdimm_redis.py:32]: Should we increase the time out a bit here ?

Comment (huangyum) [reply qemu/tests/nvdimm_redis.py:32]: Did you hit any timeout issue here? The current one works well on x86. Please let me know if you prefer a new one. Thanks.

Comment (MiriamDeng) [reply qemu/tests/nvdimm_redis.py:32]: How about set it as 600 ?

**Comment by MiriamDeng** [comment]
The patch should work well but there's test failure on ppc,it's about test suite's problem if I wasn't wrong, I need to check if we need to set extra paramaters before compliing the test suite on ppc,  will update this patch soon once i have them, thanks.

File: qemu/tests/nvdimm_redis.py (hunk 4 above)
Comment (huangyum) [reply qemu/tests/nvdimm_redis.py]: Sure, updated.

**Comment by MiriamDeng** [comment]
LGTM from code's perspective and the issue will be filed in the bug, thanks.

File: qemu/tests/nvdimm_redis.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            if s:
+                test.error("Failed to run cmd '%s', output: %s" % (cmd, o))
+        error_context.context("Run redis test in guest", logging.info)
+        s, o = session.cmd_status_output(params["run_test"], timeout=3600)
+        if s:
+            test.fail("Run redis test failed, output: %s" % o)
+        vm.verify_kernel_crash()
+    finally:
+        if session:
+            session.cmd_output_safe('rm -rf params["redis_dir"]')
```
Comment (yanan-fu) [inline qemu/tests/nvdimm_redis.py:42]: `prams["redis_dir"]` is only a string here.

Comment (huangyum) [reply qemu/tests/nvdimm_redis.py]: Updated, thx.

**Comment by yanan-fu** [comment]
Call for latest test result ~   Others LGTM

**Comment by huangyum** [comment]
(1/1) type_specific.io-github-autotest-qemu.nvdimm.nvdimm_redis: PASS (737.61 s)

**Comment by yanan-fu** [comment]
Ack

**Pull request merged**